src/System.Security.Cryptography.Algorithms/tests/CryptoConfigTests.cs

```diff
@@ -234,6 +236,57 @@ public static void CreateFromName_CtorArguments()
             Assert.Equal("Hello", ctorObj.MyString);
         }
 
+        [Fact]
+        public static void EncodeOID_Validation()
+        {
+            Assert.Throws<ArgumentNullException>(() => CryptoConfig.EncodeOID(null));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID(string.Empty));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID("BAD.OID"));
+            Assert.Throws<FormatException>(() => CryptoConfig.EncodeOID("1.2.BAD.OID"));
+            Assert.Throws<OverflowException>(() => CryptoConfig.EncodeOID("1." + uint.MaxValue));
+            CryptoConfig.EncodeOID("-1.2.-3"); // Negative values not checked
+        }
+
+        [Fact]
+        public static void EncodeOID_Length_Boundary()
+        {
+            string valueToRepeat = "1.1";
+
+            // Max size is 0x7f
+            string s = new StringBuilder(valueToRepeat.Length * 0x80).Insert(0, valueToRepeat, 0x80).ToString();
```

Comment:
So... `1.11.11.11.11.11.`...`.11.1` having 0x80 '.'s?

I think that the comments could use some improvement (and debate if the repeated 0x80/0x7F should be pulled out/named).

```C#
// Build a string like 1.11.11.11. ... .11.1, which has 0x80 separators.
// The BER/DER encoding of an OID has a minimum number of bytes as the number of separator characters,
// so this would produce an OID with a length segment of more than one byte, which EncodeOID can't handle.
//
// After verifying the exception for this case, try again with one less separator for the boundary case, but the
// particular output is really long and would just clutter up this test, so only verify it doesn't throw.
```